app/controllers/chats_controller.rb

```diff
@@ -7,11 +7,10 @@ def create
     @sender = current_user
     @recipient = User.find_by(email: chat_params[:email])
     if Chat.between(@sender.id, @recipient.id).present?
-      @chat = Chat.between(@sender.id, @recipient.id)
+      @chat = Chat.between(@sender.id, @recipient.id).first
     else
       @chat = Chat.create({sender_id: @sender.id, recipient_id: @recipient.id})
     end
```

Comment:
In this case that logic in the controller has already been merged into master- this is a fixed version, as we discovered after the first merge that it needed a .first to get out the chat object.

As far as I can tell from the unit testing Eoin and I have been attempting, we can't test the functionality of saving a chat to the database and then retrieving it because a chat can't be saved without valid users to relate to, and that requires either creating users in the database in the context of the chat unit tests (which I'm not at all sure how to do because it's all hidden in Devise), or otherwise having meaningful users in the test database to draw on, which is where we'd need factories. If there's another way of doing it I'm happy to do that!